mobile/library/common/jni/jni_helper.cc

```diff
@@ -0,0 +1,226 @@
+#include "library/common/jni/jni_helper.h"
+
+#include "source/common/common/assert.h"
+
+namespace Envoy {
+namespace JNI {
+
+jmethodID JniHelper::getMethodId(jclass clazz, const char* name, const char* sig) {
+  jmethodID method_id = env_->GetMethodID(clazz, name, sig);
+  abortOnException("Failed calling GetMethodID.");
```

Comment:
This is a very good question and something that I've been wondering about, too. Initially, I implemented by rethrowing the exception to Java, however most of these errors are unrecoverable, such as class not found, typo in the method signature, etc. For these types of errors, I think aborting is the best course of action. However, with `Call` APIs, it may make sense to not abort since in Java some exceptions are recoverable. For our use case right now, most of the exceptions are unrecoverable, but this is definitely an area that we want to revisit again. Thanks for bringing this up!

Reply:
That makes sense. I figured for most of these, it probably means a JVM mismatch, or something else which necessitates aborting the program.  I wonder if we may want to remove the abortOnException in https://github.com/envoyproxy/envoy/pull/30641/files#diff-da15bd9061d8a50cf40286a7020ebfabceadabd6a6c280c3691a89971f0276abR142 right now, since the method already requires a methodID (and presumably, if the method wasn't found, we'd abort on the getMethodId call), so the only other exceptions would presumably relate to the args (e.g. illegalargumentexception, etc).  I'm only concerned about apps crashing when they previously didn't

Reply:
Good point about crashing. I think there are potential scenarios where an app might have the ability to use EM or a different network stack, and can decide to fail over from EM if it notices that it's not working. But an app can't do that if the app crashes. So we should probably think about ways to avoid new crashes in misconfigured app scenarios.

Reply:
> That makes sense. I figured for most of these, it probably means a JVM mismatch, or something else which necessitates aborting the program. I wonder if we may want to remove the abortOnException in https://github.com/envoyproxy/envoy/pull/30641/files#diff-da15bd9061d8a50cf40286a7020ebfabceadabd6a6c280c3691a89971f0276abR142 right now, since the method already requires a methodID (and presumably, if the method wasn't found, we'd abort on the getMethodId call), so the only other exceptions would presumably relate to the args (e.g. illegalargumentexception, etc). I'm only concerned about apps crashing when they previously didn't

When calling the `Call` API, the exception still needs to be handled, i.e. calling `ExceptionCheck()`. This is one of the checks that `-Xcheck:jni` does when enabled. I think we have a few options.
- Always rethrow it to Java
- Log it and do nothing (probably not a good idea)
- Log it and abort (this is what the code does currently)
- Introduce a flag whether or not to abort or rethrow

Reply:
That seems reasonable to me too.

Reply:
Done, updated the PR to rethrow. We probably need to revisit this once we start integrating it into the existing code. I'll try to update the code piecemeal, so that we can keep the existing behavior.